lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/launch/JavaLaunchCommandBuilder.java

```diff
@@ -25,17 +25,30 @@
 import java.io.File;
 import java.util.ArrayList;
 import java.util.Collection;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
+import java.util.logging.Logger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 public final class JavaLaunchCommandBuilder implements LaunchCommandBuilder {
+  private static final Logger LOG = Logger.getLogger(JavaLaunchCommandBuilder.class.getName());
+
   private static final String DEFAULT_JAVA_PATH = System.getenv("JAVA_HOME") + "/bin/" + "java";
   private String stderrPath = null;
   private String stdoutPath = null;
-  private int megaBytes = 0;
   private String evaluatorConfigurationPath = null;
   private String javaPath = null;
   private String classPath = null;
   private Boolean assertionsEnabled = null;
+  private Map<String, JVMOption> options = new HashMap<>();
+  private Map<String, JVMOption> defaultOptions = new HashMap<String, JVMOption>() {{
```

Comment:
I don't know what the conventions are in REEF, but personally I'm uncomfortable with double-braces initialization since it introduces a hidden reference (to the enclosing object) which can't be GC'ed as long as the initialized object is alive. That's obviously not an issue in this context, but could be elsewhere. It also prevents you from using the diamond operator syntax, which isn't a huge deal, just means a little extra boilerplate. Guava has nice syntax for building immutable maps, which can be used to initialize mutable maps. Anyway, comes down to project conventions, so the committers should decide :)